src/Symfony/Bundle/SecurityBundle/Security/ContextAwareFirewallMapInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Security;
+
+use Symfony\Component\Security\Http\FirewallMapInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * This interface must be implemented by firewall maps that are able to return a
+ * context.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+interface ContextAwareFirewallMapInterface extends FirewallMapInterface
```

Comment:
I think the naming problem comes from the existing `FirewallContext::getContext()` which returns the listeners returned by `FirewallMapInterface::getListeners()`, it's confusing. The interface was here to differentiate the `FirewallMap` of the FrameworkBundle (impacted here) from the one of the Security component. After all I think it is not needed, the profiler being depending of the framework, there is no risk for accessing the security's map instead. Now I think to:

- make the `FirewallMap::getContext(): FirewallContext()` introduced here private 
- add a `FirewallMap::getConfig(): FirewallConfig` (framework) without any interface

What do you think?